src/target/riscv/debug_defines.c

```diff
@@ -0,0 +1,4194 @@
+			.lsb = 0x10,
+			.msb = 0x10,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_error5
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_full5(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "full5",
+			.lsb = 0xf,
+			.msb = 0xf,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_valid5
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_error4(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "error4",
+			.lsb = 0xe,
+			.msb = 0xe,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_full5
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_valid4(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "valid4",
+			.lsb = 0xd,
+			.msb = 0xd,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_error4
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_full4(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "full4",
+			.lsb = 0xc,
+			.msb = 0xc,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_valid4
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_error3(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "error3",
+			.lsb = 0xb,
+			.msb = 0xb,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_full4
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_valid3(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "valid3",
+			.lsb = 0xa,
+			.msb = 0xa,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_error3
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_valid0(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "valid0",
+			.lsb = 1,
+			.msb = 1,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_valid3
+	};
+	return result;
+}
+riscv_debug_reg_field_list_t dmi_sercs_get_full0(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "full0",
+			.lsb = 0,
+			.msb = 0,
+			.values = NULL
+		},
+		.get_next = dmi_sercs_get_valid0
+	};
+	return result;
+}
+
+riscv_debug_reg_field_list_t dmi_sertx_get_data(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "data",
+			.lsb = 0,
+			.msb = 0x1f,
+			.values = NULL
+		},
+		.get_next = NULL
+	};
+	return result;
+}
+
+riscv_debug_reg_field_list_t dmi_serrx_get_data(riscv_debug_reg_ctx_t context)
+{
+	riscv_debug_reg_field_list_t result = {
+		.field = {
+			.name = "data",
+			.lsb = 0,
+			.msb = 0x1f,
+			.values = NULL
+		},
+		.get_next = NULL
+	};
+	return result;
+}
+
+riscv_debug_reg_info_t get_riscv_debug_reg_info(enum riscv_debug_reg_ordinal reg_ordinal)
+{
+	switch (reg_ordinal) {
```

Comment:
Can this be an index into an array instead of a giant switch statement?

Reply:
Addressed.